src/bin/cli/mod.rs

```diff
@@ -218,10 +449,50 @@ pub(crate) fn target_arg(target_arg: &TargetArg) -> Target {
     target
 }
 
-pub fn imports_arg(package: &Package) -> FileResolver {
+pub trait PackageTrait {
+    fn get_input(&self) -> &Vec<PathBuf>;
+    fn get_import_path(&self) -> &Option<Vec<PathBuf>>;
+    fn get_import_map(&self) -> &Option<Vec<(String, PathBuf)>>;
+}
+
+impl PackageTrait for CompilePackage {
+    fn get_input(&self) -> &Vec<PathBuf> {
+        match &self.input {
```

Comment:
Could use `map_err` instead of the match here

Reply:
I think `.expect()` would look better

Reply:
Panicking is not the same as `std::process::exit(1)`. While technically the result is almost the same (I'm not sure whether we care about the different exit code). An unwrap or expect implies that this is a situation we do not anticipate for, and if it should ever happen regardless there's nothing to do about it. Here we can terminate in a more graceful way.

Reply:
but a map_err would imply converting the Option<> to a Result<> beforehand. I think an `if let Some()` would suffice, what do you think?

Reply:
Right, I didn't pay attention close enough, sorry. Yes you could return inside the `if let` and exit afterwards. Or leave it like it is